python/AMI_NOT_PUBLIC_CHECK/AMI_NOT_PUBLIC_CHECK.py

```diff
@@ -0,0 +1,406 @@
+ASSUME_ROLE_MODE = False
+
+#############
+# Main Code #
+#############
+
+# Generates non-compliant response
+def generate_evaluation_list(images, event):
+    evaluations = []
+    # Looping through all available images
+    for image in images:
+        evaluation = build_evaluation(
+            image['ImageId'],
+            "NON_COMPLIANT",
+            event,
+            resource_type=DEFAULT_RESOURCE_TYPE,
+            annotation="Amazon Machine Image Id: {} is public".format(image['ImageId'])
+        )
+        evaluations.append(evaluation)
+    return evaluations
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+
+    ec2_client = get_client('ec2', event)
+    public_ami_result = ec2_client.describe_images(
+                            Filters=[
+                                {
+                                    'Name': 'is-public',
+                                    'Values': ['true']
+                                }
+                            ],
+                            Owners=[json.loads(event['invokingEvent'])['awsAccountId']]
+                        )
+    # Check if API call was unsuccessful
+    if public_ami_result['ResponseMetadata']['HTTPStatusCode'] != 200:
+        return build_internal_error_response("Unexpected error while completing API request")
+    # If public_ami_list is empty, generate non-applicable response
+    if not public_ami_result['Images']:
+        return build_evaluation("N/A", "NOT_APPLICABLE", event, resource_type=DEFAULT_RESOURCE_TYPE)
```

Comment:
`return None`

it will do the right thing

Reply:
Done